command/agent/agent.go

```diff
@@ -544,5 +551,5 @@ func (a *Agent) syncAgentServicesWithConsul(clientHttpAddr string, serverHttpAdd
 		return host, p
 	})
 
-	return cs.SyncServices(services)
+	return a.consulSyncer.SyncServices(services)
```

Comment:
This method should not show any errors if fingerprinting failed. It leads to bad UX when there is no client.

```
* Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
* Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused    2016/05/26 17:59:09 [DEBUG] client: periodically checking for node changes at duration 5s
```